lib/misc.c

```diff
@@ -51,6 +51,9 @@ M0_INTERNAL int m0_uint128_cmp(const struct m0_uint128 *u0,
 M0_INTERNAL int m0_uint128_sscanf(const char *s, struct m0_uint128 *u128)
 {
 	int rc = sscanf(s, U128I_F, U128_S(u128));
+	if (rc == 2 && (u128->u_hi == LLONG_MAX ||
+	                u128->u_lo == LLONG_MAX))
+		rc = sscanf(s, U128X_F, U128_S(u128));
```

Comment:
Won't it be better to use PRIu64 instead of PRIi64 in U128I_F? After all, it is _unsigned_ 128-bit integer, so why do we allow something like "-1:-2"?

Reply:
`PRIu64` allows only decimal format, but we want to be able to parse in hex format also (if 0x prefix is present).

Reply:
Agree. Approved.